relay-server/src/actors/outcome.rs

```diff
@@ -848,11 +920,45 @@ impl Handler<TrackRawOutcome> for OutcomeProducer {
             }
             ProducerInner::AsHttpOutcomes(ref producer) => {
                 Self::send_outcome_metric(&message, "http");
-                producer.do_send(message);
+                let _ = producer.send(message);
                 Ok(())
             }
             ProducerInner::AsClientReports(_) => Ok(()),
             ProducerInner::Disabled => Ok(()),
         }
     }
 }
+
+impl Service for OutcomeProducer {
+    type Messages = OutcomeProducerMessages;
+}
+
+impl ServiceMessage<OutcomeProducer> for TrackOutcome {
+    type Response = Result<(), OutcomeError>;
+
+    fn into_messages(self) -> (OutcomeProducerMessages, oneshot::Receiver<Self::Response>) {
+        let (tx, rx) = oneshot::channel();
+        (OutcomeProducerMessages::TrackOutcome(self, tx), rx)
+    }
+}
+
+impl ServiceMessage<OutcomeProducer> for TrackRawOutcome {
+    type Response = Result<(), OutcomeError>;
+
+    fn into_messages(self) -> (OutcomeProducerMessages, oneshot::Receiver<Self::Response>) {
+        let (tx, rx) = oneshot::channel();
+        (OutcomeProducerMessages::TrackRawOutcome(self, tx), rx)
+    }
+}
+
+#[derive(Debug)]
+pub enum OutcomeProducerMessages {
+    TrackOutcome(TrackOutcome, oneshot::Sender<Result<(), OutcomeError>>),
+    TrackRawOutcome(TrackRawOutcome, oneshot::Sender<Result<(), OutcomeError>>),
+}
+
+impl Default for OutcomeProducer {
```

Comment:
All the `Default` implementations are no longer needed without the Actix registry.